packages/preview-middleware-client/src/adp/quick-actions/fe-v2/lr-enable-table-filtering.ts

```diff
@@ -0,0 +1,87 @@
+import { getRelevantControlFromActivePage } from '../../../cpe/quick-actions/utils';
+import {
+    GRID_TABLE_TYPE,
+    M_TABLE_TYPE,
+    SMART_TABLE_TYPE,
+    TableQuickActionDefinitionBase,
+    TREE_TABLE_TYPE
+} from '../table-quick-action-base';
+import { isUnsupportedUI5Version, executeToggleAction } from './utils';
+import { translateText } from '../../quick-actions/utils';
+
+export const ENABLE_TABLE_FILTERING = 'enable-table-filtering';
+
+const CONTROL_TYPES = [SMART_TABLE_TYPE, M_TABLE_TYPE, TREE_TABLE_TYPE, GRID_TABLE_TYPE];
+
+/**
+ * Quick Action for enabling table filtering using table personalization settings.
+ */
+export class EnableTableFilteringQuickAction
+    extends TableQuickActionDefinitionBase
+    implements NestedQuickActionDefinition {
+    constructor(context: QuickActionContext) {
+        super(ENABLE_TABLE_FILTERING, CONTROL_TYPES, 'QUICK_ACTION_ENABLE_TABLE_FILTERING', context);
+    }
+    isActive: boolean;
+    readonly forceRefreshAfterExecution = true;
+    isTableFilteringInPageVariantEnabled = false;
+    lsTableMap: Record<string, number> = {};
+    async initialize(): Promise<void> {
+        const isUI5VersionNotSupported = await isUnsupportedUI5Version();
+        if (isUI5VersionNotSupported) {
+            return;
+        }
+
+        let index = 0;
+        const tooltipText = await translateText(`THE_CHANGE_HAS_ALREADY_BEEN_MADE`);
+        const iconTabBarFilterMap = this.buildIconTabBarFilterMap();
+        for (const table of getRelevantControlFromActivePage(
+            this.context.controlIndex,
+            this.context.view,
+            CONTROL_TYPES
+        )) {
+            if (table) {
+                const tabKey = Object.keys(iconTabBarFilterMap).find((key) => table.getId().endsWith(key));
+                const isFilterEnabled = table.data('p13nDialogSettings').filter.visible;
+                this.children.push({
+                    label: tabKey ? `'${iconTabBarFilterMap[tabKey]}' table` : this.getTableLabel(table),
+                    enabled: !isFilterEnabled,
+                    tooltip: isFilterEnabled ? tooltipText : undefined,
+                    children: []
+                });
+                this.lsTableMap[`${this.children.length - 1}`] = index;
+                index++;
+            }
+        }
+
+        if (this.children.length > 0) {
+            this.isApplicable = true;
+        }
+        return Promise.resolve();
+    }
+
+
+    async execute(path: string): Promise<FlexCommand[]> {
+        const index = this.lsTableMap[path];
+        const table = getRelevantControlFromActivePage(this.context.controlIndex, this.context.view, CONTROL_TYPES);
+
+        const modifiedControl = table[index];
+        if (!modifiedControl) {
+            return [];
+        }
+        const command = await executeToggleAction(
+            this.context,
+            'component/settings',
```

Comment:
Looking at this path it looks like it looks like a simple than Nested action. Did you test v2 with multiple table on LRP

Reply:
yes, tested with multiple tables in LRP

Reply:
i am just saying if this a global setting for both table you dont need a Nestequick action

Reply:
change the logic and now the qa is available at page level.